fms2_io/netcdf_io.F90

```diff
@@ -1503,86 +1503,80 @@ function get_valid(fileobj, variable_name) &
       add_offset = 0._real64
     endif
 
-	 !Max and and min data values are defined by the valid_range, valid_min, and valid_max attributes if they are present. 
-	 !If the fill_value attribute is present and valid_range is not, then fill_value determines valid_data values. 
-	 !Otherwise, the missing_value attribute determines valid data_values if it is present, and valid_range fill_value attributes are not.
+	!Max and and min data values are defined by the valid_range, valid_min, and valid_max attributes if they are present. 
+    !Assign max/min values using "valid" attributes.
+    if (attribute_exists(fileobj%ncid, varid, "valid_range")) then
+      call get_variable_attribute(fileobj, variable_name, "valid_range", buffer)
+      valid%max_val = buffer(2)*scale_factor + add_offset
+      valid%has_max = .true.
+      valid%min_val = buffer(1)*scale_factor + add_offset
+      valid%has_min = .true.
+    else
+      if (attribute_exists(fileobj%ncid, varid, "valid_max")) then
+        call get_variable_attribute(fileobj, variable_name, "valid_max", buffer(1))
+        valid%max_val = buffer(1)*scale_factor + add_offset
+        valid%has_max = .true.
+      endif
+      if (attribute_exists(fileobj%ncid, varid, "valid_min")) then
+        call get_variable_attribute(fileobj, variable_name, "valid_min", buffer(1))
+        valid%min_val = buffer(1)*scale_factor + add_offset
+        valid%has_min = .true.
+      endif
+    endif
+    valid%has_range = valid%has_min .or. valid%has_max
```

Comment:
If there is a range, only one of these has to be defined?  It seems like this should be an `.and.`.  Shouldn't valid%has_range be set to .true. in the if block under 1508?

Reply:
I suppose this could be a topic of discussion: does a range exist if it is "open" (aka "undefined") on one end? My interpretation was that an open range should exist (and be tested). The old is_valid test only tested for a closed range, but if we're defining a minimum value, shouldn't we test to see if the values are above that minimum, regardless of whether a maximum is defined? The choice of setting `valid%has_range` to `.true.` when has_min is defined but has_max isn't (or v.v.) is also open to discussion, in other words, should `valid%has_range` be a strict "closed definition" or a more "open definition"? I tend to prefer the latter. Under that definition, I wanted to wait until those scenarios were tested (1515 and 1520 blocks) before assigning `valid%has_range`.